### PR TITLE
Update ui on round edit flow in manager

### LINKED ISSUE #2522: Manager: bugs + UI updates epic for Round Edit flow
Follow up ticket to: https://github.com/gitcoinco/grants-stack/issues/2374 

DG round on Optimism: 
QF round on Optimism: https://manager.gitcoin.co/#/round/0xa02954613075ea89d6fca6f59168ee67bf9d65d7

Tasks: 

**General**
- [x] (For both QF & DG rounds) Update copy at top of Round Settings tab to [this](https://www.figma.com/file/UIXKYToUBeyb47mVLk7atW/Manager-S19?type=design&node-id=130-2470&mode=design&t=J88ttjbl5U0CsZPn-0)
- [x] When not in edit mode, the text in all fields should be light gray (see the text color of Application Start Date for QF rounds - we want that but for all fields) 

**Round & Application Period tab**
- [x] Update Application End Date to YYYY / MM / DD when in Edit mode (see below for current state) 

Screen Shot 2023-10-30 at 9 08 28 AM

- [x] Past dates text should be light gray (see below for current UI with Round Start Date when it's a date in the past, but the text is black instead of light gray like the Application Start Date) 
Screen Shot 2023-10-30 at 9 08 28 AM 

- [x] For QF rounds, we should be able to check/uncheck rolling applications just like in round setup flow 

- [x] For DG rounds - for rounds that have "No Round End Date", I'm not able to edit the round end date to an earlier date.

### PR DESCRIPTION
**Pull request by bhargavaparoksham**
## PR checklist

For every PR, make sure that these statements are true:
- [x] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [x] Doesn't contain type casts and non-null assertions.
- [x] Doesn't add `@ts-ignore`.
- [x] Doesn't disable lints.
- [x] Doesn't use `useState` just for computation - use plain variables instead.
- [x] Splits components into pure components that don't depend on external state or hooks.
- [x] Avoid embedding components within other components
- [x] Doesn't propagate optional values without good reason, doesn't mark property values as optional if that doesn't represent reality.
- [x] Doesn't duplicate existing code.
- [x] Parses out-of-domain data - this includes user input, API respones, on-chain data etc.
- [x] Doesn't contain commented out code.
- [x] Doesn't contain skipped or empty tests.
- [x] If this PR adds/updates any feature, it adds/updates its test script

Subjective - at the discretion of the reviewers
- Does things as simply as possible, but not simpler.
- Doesn't reinvent the wheel or create premature abstractions.

##### Description

Update ui on round edit flow in manager

##### Refers/Fixes

fixes #2522

### CODE REVIEW

**Comment by melissa-neira** [comment]
@bhargavaparoksham I'm not seeing the correct date/time format here
Screenshot 2023-11-07 at 16 28 36
it should be yyyy/mm/dd

**Comment by michellema1208** [comment]
hey @bhargavaparoksham - thank you for pulling this together so quickly! Sharing testing results below: 

Links: 
* Test QF round: https://manager-git-manager-2522-grants-stack.vercel.app/#/round/0xa02954613075ea89d6fca6f59168ee67bf9d65d7 
* Test DG round: 

**All**

- [ ] When in Readonly mode, the text should be light gray. See [Figma](https://www.figma.com/file/UIXKYToUBeyb47mVLk7atW/Manager-S19?type=design&node-id=130-2470&mode=design&t=v60cv0auM0Bj2G6X-0)
> This is currently correct on the Round & Application Period tab, but not correct on the Round Details & Funding Settings tab

**Round & Application Period tab**

- [ ] +1 to Melissa's comment above - the copy and date formatting should be in YYYY / MM / DD format (it looks like it's currently in MM / DD / YYYY format) 

- [ ] If the round had rolling applications checked previously, then we should show that it's checked on Round Settings page in both Edit and Readonly mode 
> For example: this [PGN round](https://manager-git-manager-2522-grants-stack.vercel.app/#/round/0xc25a83e89c741ddb1f885db09310fce1944d58da) was most likely created with rolling applications checked, but the checkbox is not checked on the Round Settings page 

- [ ] For QF rounds where the Application End Date has not passed, when in Edit mode, the Application End Date syntax is off (try with [this PGN round](https://manager-git-manager-2522-grants-stack.vercel.app/#/round/0xc25a83e89c741ddb1f885db09310fce1944d58da))
Screen Shot 2023-11-07 at 2 11 06 PM

**Review by michellema1208 — APPROVED** [review]
LGTM - amazing work as usual, Bhargav!

**Review by melissa-neira — APPROVED** [review]
lgtm!

File: packages/round-manager/src/features/round/ViewRoundSettings.tsx [hunk 1]
```diff
@@ -122,6 +122,43 @@ export default function ViewRoundSettings(props: { id?: string }) {
...
+      setEditedRollingApplications(true);
+    }
+  }, [editedRound?.applicationsEndTime, editedRound?.roundEndTime]);
+
+  useEffect(() => {
+    if (isDirectRound(round!)) {
+      setEditedRollingApplications(true);
+    }
+  }, [round]);
+
```
Comment (carlbarrdahl) [inline packages/round-manager/src/features/round/ViewRoundSettings.tsx:161]: Could be a good idea to move this into a hook. If a lot of these states are form states we might want to add react-hook-form to handle that for us.

File: packages/round-manager/src/features/round/ViewRoundSettings.tsx [hunk 2]
```diff
@@ -861,7 +903,7 @@ function DetailsPage(props: {
                           `roundMetadata.eligibility.requirements.${i}.requirement`
                         )}
                         type="text"
-                        className="w-full rounded-md border border-gray-300 shadow-sm py-2 px-3 bg-white text-sm leading-5 focus:outline-none focus:ring-blue-500 focus:border-blue-500 transition duration-150 ease-in-out disabled:bg-gray-50"
+                        className="w-full rounded-md border border-gray-300 shadow-sm py-2 px-3 bg-white text-sm leading-5 focus:outline-none focus:ring-blue-500 focus:border-blue-500 transition duration-150 ease-in-out disabled:bg-gray-50 disabled:text-gray-400"
```
Comment (carlbarrdahl) [inline packages/round-manager/src/features/round/ViewRoundSettings.tsx:906]: Lot's of repetition here, why not create an Input component?

File: packages/round-manager/src/features/round/ViewRoundSettings.tsx [hunk 3]
```diff
@@ -888,37 +930,37 @@ function DetailsPage(props: {
...
                                 },
-                              });
-                            }}
-                          >
-                            <Cross color="red" />
-                          </button>
-                        )}
+                              },
+                            });
+                          }}
```
Comment (carlbarrdahl) [inline packages/round-manager/src/features/round/ViewRoundSettings.tsx:959]: If the logic in the top of ViewRoundSettings is moved to a hook, this logic could be added to a function.

**Review by carlbarrdahl — APPROVED** [review]
Left some suggestions to move logic into a hook. I'm not familiar with this codebase but I think it looks good otherwise.

Moment is also legacy and could be a good idea to look into replacing with date-fns or something else.

File: packages/round-manager/src/features/round/ViewRoundSettings.tsx (hunk 1 above)
Comment (bhargavaparoksham) [reply packages/round-manager/src/features/round/ViewRoundSettings.tsx:161]: Will be fixed as part of round edit flow code refactoring ticket.

**Pull request merged**